inference-engine/tests/functional/plugin/shared/include/single_layer_tests/logical.hpp

```diff
@@ -34,10 +34,20 @@ class LogicalLayerTest : public testing::WithParamInterface<LogicalTestParams>,
     virtual public LayerTestsUtils::LayerTestsCommon {
 protected:
     InferenceEngine::Blob::Ptr GenerateInput(const InferenceEngine::InputInfo &info) const override;
+    void SetupParams();
     void SetUp() override;
 
 public:
     static std::string getTestCaseName(testing::TestParamInfo<LogicalTestParams> obj);
     static std::vector<LogicalParams::InputShapesTuple> combineShapes(const std::map<std::vector<size_t>, std::vector<std::vector<size_t >>>& inputShapes);
+
+protected:
+    LogicalParams::InputShapesTuple inputShapes;
+    InferenceEngine::Precision inputsPrecision;
```

Comment:
`inputsPrecision` must be member of parent class. (see `InferenceEngine::Precision inPrc`) This may be fixed later although.